test/mpi/test_case.cc

```diff
@@ -191,10 +210,14 @@ TestCase::~TestCase()
         UCC_CHECK(ucc_collective_finalize(req));
     }
     if (sbuf) {
-        UCC_CHECK(ucc_mc_free(sbuf_mc_header));
```

Comment:
I think "if (sbuf)" is no longer needed, you can just leave "if (sbuf_mc_header)". rbuf similarly

Reply:
done